src/core/features/course/pages/module-preview/module-preview.ts

```diff
@@ -92,6 +96,42 @@ export class CoreCourseModulePreviewPage implements OnInit {
         this.loaded = true;
     }
 
+    /**
+     * Opens a module summary page.
+     */
+    async openModuleSummary(): Promise<void> {
+        if (!this.module) {
+            return;
+        }
+
+        const data = await CoreDomUtils.openSideModal<CoreCourseModuleSummaryResult>({
+            component: CoreCourseModuleSummaryComponent,
+            componentProps: {
+                moduleId: this.module.id,
+                module: this.module,
+                description: this.module.description,
+                component: this.module.modname,
+                courseId: this.courseId,
+                displayOptions: {
+                    // Show description on summary if not shown on the page.
+                    displayDescription: false,
+                    displayBlog: false,
+                },
+            },
+        });
+
+        if (data) {
+            if (!this.loaded && (data.action == 'refresh' || data.action == 'sync')) {
```

Comment:
```suggestion
            if (this.loaded && (data.action == 'refresh' || data.action == 'sync')) {
```

Shouldn't this be the opposite? We want to refresh if data is loaded. Also, can we have a 'sync' action in module preview?

Reply:
true... :-(